### PR TITLE
fix: chart import error with virtual dataset

### PR DESCRIPTION
**Pull request by codemaster08240328**
### SUMMARY

This issue was not related to importing/exporting over api.

This was happened through UI as well, and the reason was that we regards virtual dataset as sql lab view.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

BEFORE:
https://github.com/apache/superset/issues/13237

AFTER:
[[DEV] Superset - 19 April 2022 - Watch Video](https://www.loom.com/share/6b72207ac26840d8afd9a9b343e72af2)

### TESTING INSTRUCTIONS

Check: https://github.com/apache/superset/issues/13237

### ADDITIONAL INFORMATION

- [X] Has associated issue: https://github.com/apache/superset/issues/13237
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by diegomedina248 — APPROVED** [review]

**Comment by rusackas** [comment]
Not sure why that logic was there in the first place.... @diegomedina248 or @zhaoyongjie do you see any risks here we're not aware of?

File: superset/charts/commands/importers/v1/__init__.py [hunk 1]
```diff
@@ -89,9 +89,7 @@ def _import(
                 config.update(
                     {
                         "datasource_id": dataset.id,
-                        "datasource_type": "view"
-                        if dataset.is_sqllab_view
-                        else "table",
+                        "datasource_type": "table",
```
Comment (zhaoyongjie) [inline superset/charts/commands/importers/v1/__init__.py:92]: I am not familiar with this part, but the `datasource_type = view` seems the correct value here. Here is the schema [defination](https://github.com/apache/superset/blob/22a92ed72278d45049cc33eba5219d215625e4d9/superset/charts/schemas.py#L199-L203) for updating chart.

@betodealmeida could you take a look?

Comment (codemaster08240328) [reply superset/charts/commands/importers/v1/__init__.py:92]: but virtual dataset should have `view` as a datasource_type?

Comment (betodealmeida) [reply superset/charts/commands/importers/v1/__init__.py:92]: I'm also not familiar with this part, so I created a chart from a virtual dataset and it has `datasource_type` set to `table`. Looking at the source code it seems that "view" here means a database `VIEW`, not a virtual dataset, and is only used by the DB engine spec to retrieve the list of views from a given database:

https://github.com/apache/superset/blob/4a835a4299bbe90def232e376f919bc494b2d0a1/superset/db_engine_specs/base.py#L873-L879

So I think this should be `table`, and we also need to fix this line:

https://github.com/apache/superset/blob/4a835a4299bbe90def232e376f919bc494b2d0a1/superset/commands/importers/v1/examples.py#L141

**Review by betodealmeida — APPROVED** [review]
This looks consistent with the current logic, even if it doesn't make sense to me. Can you also fix `commands/importers/v1/examples.py` in this PR? Thanks!

**Pull request merged**